lib/Site.js

```diff
@@ -183,10 +187,23 @@ Site.prototype.createPageData = function (config) {
   const sourcePath = path.join(this.rootPath, config.pageSrc);
   const tempPath = path.join(this.tempPath, config.pageSrc);
   const resultPath = path.join(this.outputPath, setExtension(config.pageSrc, '.html'));
+
+  const faviconPathWithoutRoot = (config.faviconPath !== undefined
+    ? config.faviconPath
+    : FAVICON_DEFAULT_PATH);
+  const faviconPath = url.join(config.baseUrl, faviconPathWithoutRoot);
+
+  if (!fs.existsSync(path.join(this.rootPath, faviconPathWithoutRoot))) {
+    if (config.faviconPath !== undefined) {
```

Comment:
Hmm, the logic is not obvious here.
We don't want to set `FAVICON_DEFAULT_PATH` if it doesn't exist.
Let's name the processed variable as `faviconUrl`.

How about:

```js
let faviconUrl;
if (config.faviconPath) {
  faviconUrl = url.join(config.baseUrl, config.faviconPath);
  if (!fs.existsSync(path.join(this.rootPath, config.faviconPath))) {
    logger.warn(`${config.faviconPath} does not exist`);
  }
} else if (fs.existsSync(path.join(this.rootPath, FAVICON_DEFAULT_PATH))) {
  faviconUrl = url.join(config.baseUrl, FAVICON_DEFAULT_PATH);
}
```